cookbook/bundles/tests.md

```diff
@@ -0,0 +1,268 @@
+.. index::
+   single: Bundle; Testing Bundles
+
+Writing tests for third party bundles
+=====================================
+
+Let's have a look at a minimized version og the Standard Edition's front controller:
+
+    <?php
+    // File: web/app.php
+
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/../vendor/autoload.php';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    // Create the request from PHP variables super globals such as $_GET, $_POST, $_SERVER, etc
+    $request = Request::createFromGlobals();
+
+    // The application handles the Request
+    $kernel = new AppKernel('prod', false);
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+As we can see Symfony applications mimick the HTTP protocol, which means that we can
+test them by building "manually" a ``Request``, giving it to our application and then
+check the returned ``Response``.
+
+In this article, we'll see how to write such tests in third party bundles.
+
+Creating a minimal AppKernel
+----------------------------
+
+To be able to test our third party bundle, we'd need an application. This issue
+can be solved by creating a minimal ``AppKernel`` inside our bundle, for testing
+or showcase purpose only.
+
+Let's write one:
+
+    <?php
+    // File: Tests/app/AppKernel.php
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            return array(
+                new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+                // register our bundle here
+            );
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load(__DIR__.'/config.yml');
+        }
+    }
+
+.. note::
+
+    With this file we add a direct dependency on Symfony's FrameworkBundle, so we need
+    to run the folowing command: `composer require --dev symfony/framework-bundle`
+
+In its minimum state `AppKernel` requires one configuration parameter, `secret`:
+
+    # File: Tests/app/config.yml
+    framework:
+        secret: Th1s1sS3cr3t!
+
+If we want to use this kernel in our functional tests, we'll need to load it:
+
+    <?php
+    // File: Tests/app/autoload.php
+
+    $loader = require __DIR__.'/../../vendor/autoload.php';
+    require __DIR__.'/AppKernel.php';
+
+Finally, running our application will create logs and cache directories, so we want to ignore them:
+
+    # File: .gitignore
+    /Tests/app/cache
+    /Tests/app/logs
+
+Running commands
+----------------
+
+Let's pretend we created a command in our bundle. we'd like to run it just to
```

Comment:
Well, we avoid the first person perspective in general. So most of this article needs to be rewritten with that in mind.